src/seacoral-cbmc/runner.ml

```diff
@@ -150,47 +149,56 @@ let uncovered_properties
   SimpleLabelEnv env
 
 
-(* [treat_cbmc_output_stream stream] reads the stream contents, expecting it to
-   be an array of json objects (the output format of CBMC), and returns the
-   corresponding stream of json objects. *)
-let treat_cbmc_output_stream stream =
-  let json = Buffer.create 42 in
-  let fmt = Format.formatter_of_buffer json in
-  let parenthesis_depth = ref 0 in
-  let junk = ref false in
-  Lwt_stream.map_list begin fun l ->
-    let l_no_space = String.trim l in
-    if l_no_space = "" || !junk then [] else
-      match l_no_space.[0] with
-      | '[' when !parenthesis_depth = 0 ->
-          parenthesis_depth := 1;
-          []
-      | _ ->
-          let elements = ref [] in
-          String.iter begin function
-            | '{' | '[' as c ->
-                incr parenthesis_depth;
-                Format.pp_print_char fmt c;
-            | ']' when !parenthesis_depth = 1 ->
-                junk := true
-            | '}' | ']' as c ->
-                decr parenthesis_depth;
-                Format.pp_print_char fmt c;
-                if !parenthesis_depth = 1 then begin
-                  Format.pp_print_flush fmt ();
-                  let j = Buffer.contents json in
-                  if !junk
-                  then Log.debug "Internal@ warning:@ ignored@ garbage@;%s" j
-                  else elements := j :: !elements;
-                  Buffer.clear json;
-                end
-            | ',' when !parenthesis_depth = 1 ->
-                ()
-            | c ->
-                Format.pp_print_char fmt c
-          end l_no_space;
-          List.rev !elements
-  end stream
+(* [treat_cbmc_output_stream encoding stream] reads the stream contents,
+   expecting it to be an array of json objects (the output format of CBMC), and
+   returns the corresponding stream of json objects decoded according to
+   [encoding]. *)
+let decode_cbmc_output_stream encoding stream =
+  let out, emit = Lwt_stream.create () in
+  Lwt.async begin fun () ->
+    let json = Buffer.create 42 in
+    let parenthesis_depth = ref 0
+    and in_quotes = ref false
+    and junk = ref false in
+    Lwt.map ignore @@
+    Lwt_stream.fold begin fun l escaping ->
```

Comment:
All the context of the fold (parenthesis, in_quotes, junk) is handled by references, but not escaping isn't. Is there a reason for it?